esb/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/part/EsbMultiPageEditor.java

```diff
@@ -549,24 +564,40 @@ public void run() {
 	/**
 	 * Performs necessary house-keeping tasks whenever the design view is
 	 * activated.
+	 * @throws Exception
 	 */
-	private void handleDesignViewActivatedEvent() {
+	private void handleDesignViewActivatedEvent() throws Exception {
 
 //		if (null != sourceEditor.getObject()) {
 //			rebuildModelObject(objectSourceEditor.getObject());
 //		}
-		if(sourceEditor!=null){
+		if (sourceEditor != null) {
 			String xmlSource = sourceEditor.getDocument().get();
-			if(xmlSource!=null && sourceDirty){
-				  if(!xmlSource.trim().isEmpty()){
-					  rebuildModelObject(xmlSource); 
-				  }
+			if (xmlSource != null) {
```

Comment:
Check for this sourceDirty was handled in the higher level before. Any way this logic have been modified now.
Thanks